pkg/chunk/cache/memcached.go

```diff
@@ -146,19 +146,20 @@ func (c *Memcached) Fetch(ctx context.Context, keys []string) (found []string, b
 
 func (c *Memcached) fetch(ctx context.Context, keys []string) (found []string, bufs [][]byte, missed []string) {
 	var items map[string]*memcache.Item
-	err := instr.CollectedRequest(ctx, "Memcache.GetMulti", c.requestDuration, memcacheStatusCode, func(innerCtx context.Context) error {
-		sp := opentracing.SpanFromContext(innerCtx)
-		sp.LogFields(otlog.Int("keys requested", len(keys)))
+	method := "Memcache.GetMulti"
+	err := instr.CollectedRequest(ctx, method, c.requestDuration, memcacheStatusCode, func(innerCtx context.Context) error {
+		log, _ := spanlogger.New(innerCtx, method)
```

Comment:
`spanlogger.New()` starts a new span. Shouldn't we always replace `innerCtx` with the context returned by the `New()` function?

Reply:
here I should have used `spanlogger.FromContext`

Reply:
hmmm, it looks like we never actually use innerCtx in the function other than to create the span? and without using `spanlogger.New` we don't get the `method` field in log lines for free